test/test_selectors.py

```diff
@@ -681,24 +705,81 @@ def test_above_fd_setsize(self):
         self.assertEqual(limit_nofile // 2, len(s.select()))
 
 
+@skipUnlessHasSelector
+class TestUniqueSelectScenarios(BaseSelectorTestCase):
+    def test_select_module_patched_after_import(self):
+        # This test is to make sure that after import time
+        # calling DefaultSelector() will still give a good
+        # return value. This issue is caused by gevent, eventlet.
+
+        # Now remove all selectors except `select.select`.
+        self.patch_select_module('select')
+
+        # Make sure that the selector returned only uses the selector available.
+        selector = self.make_selector()
+        self.assertIsInstance(selector, selectors.SelectSelector)
+
+    @skipUnlessHasENOSYS
+    def test_select_module_defines_does_not_implement_poll(self):
+        # This test is to make sure that if a platform defines
+        # a selector as being available but does not actually
+        # implement it (kennethreitz/requests#3906) does not
+        # make DefaultSelector() fail.
```

Comment:
This comment has got some confusing grammar in it. It should probably end with `does not actually implement it (kennethreitz/requests#3906) then DefaultSelector() does not fail`.